rwt.py

```diff
@@ -0,0 +1,266 @@
+            exit()
+
+    result = learn.SetReadingUserActiveInfo(SerialNumber, LicenseNumber, ActivationCode)
+    status = result['SetReadingUserActiveInfoResult']
+    if status == '1':
+        library.success('user ' + learn.username + ' active successfully')
+    else:
+        library.fail('user ' + learn.username + ' active failed: ' + result['ReturnMessage'])
+        exit()
+    return status
+
+
+def login(learn):
+    result = learn.LoginReading()
+    status = result['LoginReadingResult']
+    if status == '1':
+        library.warning('user ' + learn.username + ' need activation')
+        exit()
+    elif status != '2':
+        library.fail('user ' + learn.username + ' login failed: ' + result['ReturnMessage'])
+        exit()
+    else:
+        library.success('user ' + learn.username + ' login successfully')
+    return status
+
+
+def get_progress(learn):
+    result = learn.GetReadingProgress()
+    status = result['GetReadingProgressResult']
+    return status
+
+
+def update_unit(learn):
+    result = learn.SetReadingUnitLearnStaus('1')
+    status = result['SetReadingUnitLearnStausResult']
+    if status == '1':
+        library.success('update unit learn status success: unit ' + learn.UnitID)
+    else:
+        library.fail('update unit learn status fail: unit ' + learn.UnitID)
+        exit()
+    return status
+
+
+def end_section(learn, score):
+    if int(score) == score:
+        score_str = "%d" % int(score)
+    else:
+        score_str = "%.1f" % score
+    result = learn.SetReadingResponseInformation(score_str, '1')
+    status = result['SetReadingResponseInformationResult']
+    if status == '1':
+        library.success('ended section ' + learn.SectionID + ' successfully: score ' + score_str)
+    else:
+        library.fail('ended section ' + learn.SectionID + ' failed')
+        # Retry login
+        status = login(learn)
+        # Retry set response
+        result = learn.SetReadingResponseInformation(score_str, '1')
+        status = result['SetReadingResponseInformationResult']
+        if status == '1':
+            library.success('ended section ' + learn.SectionID + ' successfully: score ' + score_str)
+        else:
+            library.fail('ended section ' + learn.SectionID + ' failed twice')
+            exit()
+    return status
+
+
+def start_section(learn):
+    result = learn.SetReadingResponseInformation('0', '0')
+    status = result['SetReadingResponseInformationResult']
+    if status == '1':
+        library.success('started section ' + learn.SectionID + ' successfully')
+    else:
+        library.fail('started section ' + learn.SectionID + ' failed')
+        exit()
+    return status
+
+
+if __name__ == '__main__':
+    r = learn_rwt("122.207.65.163", "USERNAME", "PASSWORD", "1")
+    r.LoginReading()
+    r.SectionID = "SECTION"
+    r.GetServerTime()
+    r.SetReadingResponseInformation('100', '1')
```

Comment:
this fixed test code should
* be removed, or
* accept cli input, or
* be commented out

I prefer the first one, but all of them are acceptable :)

Reply:
Test case removed